src/evaluator.h

```diff
@@ -142,7 +142,7 @@ struct Evaluator {
                     if (g->xs == 1 || g->ys == 1) {
                         return op->runl(g);
                     } else {
-                        std::vector<unique_ptr<Grid>> gs;
+                        std::vector<shared_ptr<Grid>> gs;

```

Comment:
I see no reason to use `shared_ptr` here.. they are way more heavyweight than `unique_ptr`, and harder to know what is going to happen.. would like to avoid them unless absolutely needed.